src/merge.ts

```diff
@@ -0,0 +1,25 @@
+import { EventPayload } from './EventPayload';
+
+/** Used for merging request JSON into an existing EventPayload. Returns a new EventPayload object */
+export const merge = (json1: EventPayload, json2: Record<string, unknown>): EventPayload => {
```

Comment:
I originally used json1 and json2 as a placeholder. We should use more descriptive naming here. Maybe original and newValues?